brave/src/main/java/brave/internal/recorder/Recorder.java

```diff
@@ -71,12 +76,14 @@ public void tag(TraceContext context, String key, String value) {
 
   /** @see brave.Span#remoteEndpoint(Endpoint) */
   public void remoteEndpoint(TraceContext context, Endpoint remoteEndpoint) {
+    if (noop.get()) return;
     if (remoteEndpoint == null) throw new NullPointerException("remoteEndpoint == null");
     spanMap.getOrCreate(context).remoteEndpoint(remoteEndpoint);
   }
 
   /** @see Span#finish() */
   public void finish(TraceContext context, long finishTimestamp) {
+    if (noop.get()) return;
```

Comment:
for these down, you don't want to early return as they remove data from the map (we want that data removed even in noop in case a span started before the tracing component was set to noop status)